### DISCUSSION TITLE
Evaluation-plan-template: issue with Concentration-Time Profiles with no observed data

### DISCUSSION THREAD

**Discussion by GhazalMontaseri** [Open-Systems-Pharmacology/Forum#1569] [Questions & Problems]
Dear OSPSuite experts, I downloaded the files available in:
https://github.com/Open-Systems-Pharmacology/Evaluation-plan-template/tree/master 

and executed it with no error. Then, I simplified the 'evaluation_plan.json' file by setting: "GOFMergedPlots": [] and removed the corresponding part in the 'Sections'. This way in the report only 'Concentration-Time Profiles' are plotted (no GoF plots). Executing the reportingEngine worked well. Then, in PK-Sim, I loaded the 'Compound.json' and removed the observed data in the 'Sim 1 - intravenous' simulation and export the updated file as 'Compound.json' and executed the createQualificationReport. However, this time I got this error: 

! Warning	argument 1 is not a vector
Configuration Plan Information:
Project: Compound
Simulation: Sim 1 - intravenous
SectionReference: ct-profiles
Plot: list(...)
PlotNumber: 0 (json numbering starting at 0);  1 (actual plot index)

	
> Error Trace
	 ↪ createQualificationReport(qualificationRunnerFolder, pkSimPortableFolder)
	  ↪ workflow$runWorkflow()
	   ↪ self[[plotTask]]$runTask(self$configurationPlan)
	    ↪ self$getTaskResults(configurationPlan, self$settings)
	     ↪ ifNotNull(timeProfilePlan$Plot$Type, plotQualificationPopulationTimeProfile(simulationAnalysis = timeProfilePlan$Plot$Analysis, 
	      ↪ plotQualificationMeanTimeProfile(configurationPlanCurves = timeProfilePlan$Plot$Curves, 
	       ↪ updateQualificationTimeProfilePlotConfiguration(simulatedMetaData = simMetaData, 
	        ↪ order(observedMetaData$id)
Error: argument 1 is not a vector
Configuration Plan Information:
Project: Compound
Simulation: Sim 1 - intravenous
SectionReference: ct-profiles
Plot: list(...)
PlotNumber: 0 (json numbering starting at 0);  1 (actual plot index)

here is the zip file:
[[Files.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/12039648/Files.zip)](https://github.com/GhazalMontaseri/Test2)

Two points: 
1) When I create a single datapoint observation at time=0 (or any time) and concentration = 0 mmol/l and add it to the simulation, it works again.
2) population simulations without observation work well, only with individual population I had the issue.

I would appreciate if you can help me with this issue?

**Comment by Yuri05** [comment]
seems to be a bug. created an issue for this: https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/1082

**Comment by pchelle** [comment]
Dear @GhazalMontaseri 
The bug has been fixed in the development version of the reporting engine.
Please find the link to the corresponding bundle here: [ospsuite.reportingengine_2.2.283.zip](https://ci.appveyor.com/api/buildjobs/ljmi27u3k38hbyta/artifacts/ospsuite.reportingengine_2.2.283.zip)

**Reply by GhazalMontaseri** [reply]
@pchelle thanks a lot for fixing the issue.

**Comment by pchelle** [comment]
While reproducing the issue from your files, I noticed you were loading the namespace of many other packages.
Since the reporting engine package is often used within the ecosystem of `{tlf}`, `{ospsuite}` and `{rClr}` packages,
loading `{ospsuite.reportingengine}` into the namespace directly loads the 3 previous libraries into the namespace (as illustrated below).
Thus, you don't need to call for these libraries in your script.

```r
library(ospsuite.reportingengine)
```
```r
#> Loading required package: ospsuite.reportingengine
#> Loading required package: ospsuite
#> Loading required package: rClr
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000
#> Loading required package: tlf
```

**Reply by GhazalMontaseri** [reply]
thanks for your comment. I removed these packages.

**Reply by GhazalMontaseri** [reply]
Dear @pchelle, I tried to execute the 'createQualificationReport' with the updated reportingEngine package and I got this error:
> Error Trace
	 ↪ createQualificationReport(qualificationRunnerFolder, pkSimPortableFolder)
	  ↪ workflow$runWorkflow()
	   ↪ self[[plotTask]]$runTask(self$configurationPlan)
	    ↪ self$getTaskResults(configurationPlan, self$settings)
	     ↪ ifNotNull(timeProfilePlan$Plot$Type, plotQualificationPopulationTimeProfile(simulationAnalysis = timeProfilePlan$Plot$Analysis, 
	      ↪ plotQualificationMeanTimeProfile(configurationPlanCurves = timeProfilePlan$Plot$Curves, 
	       ↪ updateQualificationTimeProfilePlotConfiguration(simulatedMetaData = simMetaData, 
Error: cannot add bindings to a locked environment
Configuration Plan Information:
Project: Compound
Simulation: Sim 1 - intravenous
SectionReference: ct-profiles
Plot: list(...)
PlotNumber: 0 (json numbering starting at 0);  1 (actual plot index)

**Reply by pchelle** [reply]
Could you check and let me know the version of the `{tlf}` package ?

```r
packageVersion("tlf")
```

The error is very likely due to the following breaking change in the `{tlf}` package in May: PR #[466](https://github.com/Open-Systems-Pharmacology/TLF-Library/pull/466)

In case, I also provided the link to the development version `{tlf}` package if you wish to try it: [tlf_1.5.150.zip](https://ci.appveyor.com/api/buildjobs/umpgkvpc7tqxrwbt/artifacts/tlf_1.5.150.zip)

**Reply by Yuri05** [reply]
you probably need to update another OSP packages as well:
https://ci.appveyor.com/api/buildjobs/0we3pmc0vaq98jf5/artifacts/ospsuite.utils_1.4.27.tar.gz
https://ci.appveyor.com/api/buildjobs/umpgkvpc7tqxrwbt/artifacts/tlf_1.5.150.tar.gz
https://ci.appveyor.com/api/buildjobs/u70we0dkcv4e94q3/artifacts/ospsuite_11.2.305.zip

**Reply by GhazalMontaseri** [reply]
@pchelle it is ‘1.5.125’.
